### PR TITLE
[Quant][Inductor] Enable quantization linear pattern fusion for gelu inside inductor

### PR DESCRIPTION
**Pull request by Le-Zheng**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #114101
* #114100
* #114099

**Summary**
Enable QLinear Unary pattern for gelu with int8 

**Test plan**
python test/inductor/test_mkldnn_pattern_matcher.py -k test_qlinear_unary

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler

### CODE REVIEW

**Comment by leslie-fang-intel** [comment]
Hi @Le-Zheng, please add a UT for it. Refer to: https://github.com/pytorch/pytorch/blob/7963aaac4157c961c1d54033d1f2515d0c987360/test/inductor/test_mkldnn_pattern_matcher.py#L1007

File: torch/_inductor/fx_passes/quantization.py [hunk 1]
```diff
@@ -519,6 +519,10 @@ def __init__(self, op_name: str, scalars_attr=None, algorithm_attr=None):
                 generate_pattern_with_unary(qlinear_pt2e_pattern, aten.relu.default),
                 dtype=original_pattern_output_dtype,
             ),
+            UnaryAttr("gelu", [], ""): generate_pattern_with_output_quant(
```
Comment (leslie-fang-intel) [inline torch/_inductor/fx_passes/quantization.py:522]: We need to cover both `None` and `tanh` as `algorithm_attr`, right? Should we need to register it twice with different `algorithm_attr`?

**Review by leslie-fang-intel — CHANGES_REQUESTED** [review]
I think we also need to modify the Inductor pattern match implementation. And also please add a UT for it. For example, GeLU should already be decomposed into primary operators, refer to FP32/BF16 implementation: https://github.com/pytorch/pytorch/blob/7963aaac4157c961c1d54033d1f2515d0c987360/torch/_inductor/fx_passes/mkldnn_fusion.py#L79-L124

**Comment by Le-Zheng** [comment]
This pull requested is migrated to https://github.com/pytorch/pytorch/pull/114854

**Pull request closed without merging**